scripts/gha/desktop_tester.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Runs and validates desktop C++ testapps.
+
+Usage:
+  python desktop_tester.py --testapp_dir ~/Downloads/testapps
+
+This will search --testapp_dir for files whose name is given by --testapp_name
+(default: "integration_test", or "integration_test.exe" on Windows), execute
+them, parse their log output according to the C++ test summary format, and
+report a summary of results.
+
+"""
+
+import os
+import platform
+import subprocess
+import threading
+
+from absl import app
+from absl import flags
+from absl import logging
+import attr
+
+from integration_testing import test_validation
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string("testapp_dir", None, "Look for testapps in this directory.")
+flags.DEFINE_string("testapp_name", "integration_test", "Name of the testapps.")
+
+
+def main(argv):
+  if len(argv) > 1:
+    raise app.UsageError("Too many command-line arguments.")
+
+  testapp_dir = _fix_path(FLAGS.testapp_dir)
+
+  testapp_name = FLAGS.testapp_name
+  if platform.system() == "Windows":
+    testapp_name += ".exe"
+
+  testapps = []
+  for file_dir, _, file_names in os.walk(testapp_dir):
+    for file_name in file_names:
+      if file_name == testapp_name:
+        testapps.append(os.path.join(file_dir, file_name))
+  if not testapps:
+    logging.error("No testapps found.")
```

Comment:
Might be worth logging the directory that doesn't have testapps in it, to help detect configuration issues?